### ISSUE TITLE
Design & Text for Widget

### ISSUE THREAD

**Issue by rypan** [18F/feedback-widget#2]
I bet you've already done some tinkering here. But I was looking through: https://playbook.cio.gov/designstandards/ and pulled together a mock of what the simple buttons would look like.

![image](https://cloud.githubusercontent.com/assets/442198/10211804/421dde9e-67c5-11e5-8f8d-b0aaeeef9be1.png)

In context, it would look something like:

![image](https://cloud.githubusercontent.com/assets/442198/10211812/59bda5ca-67c5-11e5-8cf5-9493a751ecb2.png)

State 1: `How was your experience?` --> `Good` or `Bad`
State 2: `How can we improve?` --> `Text Field` w. limited characters
State 3: `Thank you for your feedback.`

![image](https://cloud.githubusercontent.com/assets/442198/10211890/1720e3b6-67c6-11e5-85fc-7f84dcab5834.png)

**Comment by rypan** [comment]
Any thoughts or feels on this @vz3 & @colinpmacarthur?

**Comment by emileighoutlaw** [comment]
Hey y'all! (I love this project, btw, and have been lurking as you develop!)

As a resident content person, can I suggest we add "optional" somewhere on the "How can we improve" box? I'm a fan of as few words as possible, but in my experience, folks often give up if they think they're required to write something. Thoughts on that?

**Comment by colinpmacarthur** [comment]
Thanks for kicking this off with some great mocks, @rypan!

Couple thoughts:
1. We probably need to noodle on the wording "How was your experience" a little bit. Maybe @emileighoutlaw can help us out?
2. Thoughts about icons in the buttons? Not part of the standards (yet), but might speed the process a bit.
3. Agree with @emileighoutlaw on the making the text field optional. Also wonder whether we still capture folks :+1: or :-1: if they don't hit submit. Seems like it would be good collect that, even if they don't make it farther.

**Comment by rypan** [comment]
Definitely agree w. @emileighoutlaw and @colinpmacarthur  that **How was your experience** it should be optional. 

Also that we should be capturing folks if they don't hit submit. Thoughts on adding two options `Submit` and `Skip`?

**Comment by emileighoutlaw** [comment]
Ohh! Great idea on two options @rypan! I like `skip`, but fear it might sound like there's more to come? How about something like `no thanks` or `maybe next time` instead?

**Comment by rypan** [comment]
True. I'm thinking it _might_ work though. When they tap 'skip' they'll see the screen saying 'Thank you for your feedback.'

Would the 'skip' button go on the left or right of 'submit'? It should be the less prominent one.

**Comment by brittag** [comment]
Gov.uk has an end-of-the page feedback widget on their services explanation pages (along with a satisfaction survey after people make progress on a specific task), and they did some research that could inform these ideas! [Here's a random example page.](https://www.gov.uk/corporation-tax-accounting-period) After the end of the content, the page has a small grey link that says "Is there anything wrong with this page?", and when you click that, it expands to "What you were doing" and "What went wrong":

<img width="566" alt="screen shot 2015-10-01 at 11 08 34 am" src="https://cloud.githubusercontent.com/assets/391313/10229476/49b1e926-682d-11e5-80af-179c14d47921.png">

Their [research notes](https://designpatterns.hackpad.com/User-satisfaction-and-user-feedback-zfk4wUpWNj3#:h=We-did-some-research-into-an-a) explain they tried replacing that with "Was this page useful? [Yes / No / Not sure] How could we improve it? [Text box]" in order to get a satisfaction score along with feedback. They decided to go back to the original though, since they found "the form didn’t capture what the problem was in many case, and encouraged providing the solution".

More links about their work on satisfaction scores and feedback: ["How good is your service? How many users give up?"](https://designnotes.blog.gov.uk/2015/08/13/how-good-is-your-service-how-many-users-give-up/) (how they improved their satisfaction surveys), ["Improving content with anonymous feedback"](https://insidegovuk.blog.gov.uk/2015/06/29/improving-content-with-anonymous-feedback/) (how they help staff access the feedback), ["User satisfaction: How satisfied are the people who use your service?"](https://www.gov.uk/service-manual/measurement/user-satisfaction.html) (short guide for content owners/writers who want to get useful feedback).

**Comment by rypan** [comment]
Nice share @brittag. It's pretty awesome to see them iterate and share how their experiments are performing. You are right, we can definitely learn from them.

I really enjoyed this link you shared:
https://designnotes.blog.gov.uk/2015/08/13/how-good-is-your-service-how-many-users-give-up/

They explain how before they were only asking satisfaction questions at the completion of the funnel, but then ended up adding them at different "done" points, like error screens.

![image](https://cloud.githubusercontent.com/assets/442198/10239081/03e91fae-6896-11e5-8019-5c42293e8bdf.png)

Also, from their design research notes. It looks like they have 3 types of surveys.

```
On GOV.UK – satisfaction survey
Every 50 page-loads we insert a banner: "Tell us what you think of GOV.UK – take the 3 minute survey". This takes you to this survey: https://www.surveymonkey.com/s/6HZFSVC
The survey has 13 screens of questions including demographic information and satisfaction scores.

On GOV.UK – is there anything wrong with this page?
On every page of GOV.UK there is a link just above the footer – 'Is there anything wrong with this page?'
This opens a short form that asks you 'What you were doing' and 'What went wrong'

On services
The Service Manual requires services to link through to Done page, which asks 2 questions:
Overall, how did you feel about the service you received today?
- Very satisfied
- Satisfied
- Neither satisfied or dissatisfied
- Dissatisfied
- Very dissatisfied
How could we improve this service?
```

I think if we start simple, with two questions, we'll be able to test out an MVP in our world. I think we need to figure out the right first question. Is it:
- How was your experience? Good / Bad
- Did this service meet your needs? Yes / No

Followed by an open ended prompt.

**Comment by colinpmacarthur** [comment]
I quite like "Did this service meet your needs?" 

I wonder whether "service" is too narrow though.

How about "Did this website meet your needs?" or "Is this agency/department meeting your needs?"

**Comment by emileighoutlaw** [comment]
We want them to respond about the website, so I think "Did this website meet your needs?" is perfect. _This service_ feels a bit too Brit English to me. And _This agency_ — to my thinking — invites in a lot of other feelings, other than about the webpage.

**Comment by emileighoutlaw** [comment]
p.s. Can I just say how much I _love_ having a 10+ comment conversation about microcopy??

**Comment by rypan** [comment]
I **love** it too!

Good call on website vs. service. While we call them services, our users most likely do not.

**Comment by rypan** [comment]
What's the proper copy (that is short!) to make sure no one enters personal information into the larger text field?

**Comment by colinpmacarthur** [comment]
Good question. How about "Don't enter anything than could identify you this form?"

**Comment by rypan** [comment]
I borrowed language from CFPB for the warning. I figure they've done plenty of testing...

![image](https://cloud.githubusercontent.com/assets/442198/10261556/e3358e92-696b-11e5-9d01-7945ac6979eb.png)

**Comment by emileighoutlaw** [comment]
Love that CFPB copy!

**Comment by colinpmacarthur** [comment]
@jmcarp or @afeld: Any chance one of you can take a stab at implementing @rypan's design? 

Would be great to have a site with a working feedback widget as a departing gift for him :)

**Comment by rypan** [comment]
I plan on pitching in all the way to the end of October...then as a open source contributor after that!

@jmcarp @afeld - did you decide on an approach? Possible to just push into Google Analytics - like you suggest here: https://github.com/18F/feedback-widget/issues/5?

**Comment by colinpmacarthur** [comment]
@noahmanger and the beta FEC team has been working on a feedback widget that we might want to fold this effort into? Given the limits of my time, I'm going to step back and see how the FEC's team widget works out and let others drive this forward. :)
